src/quantuminspire/qiskit/circuit_parser.py

```diff
@@ -17,170 +17,410 @@
 """
 import copy
 import numpy as np
-from typing import Dict, Any
+from io import StringIO
+from typing import Optional
+from qiskit.qobj import QobjItem
+from quantuminspire.exceptions import ApiError
 
 
 class CircuitToString:
     """ Contains the translational elements to convert the Qiskit circuits to cQASM code."""
 
-    def _cx(self, circuit: Dict[str, Any]) -> str:
+    @staticmethod
+    def _gate_not_supported(stream: StringIO, instruction: QobjItem, binary_control: Optional[str] = None) -> None:
+        """ Called when a gate is not supported with the backend. Throws an exception (ApiError)
+
+        Args:
+            stream: Parameter not used.
+            instruction: The Qiskit instruction to translate to cQASM.
+            binary_control: Parameter not used.
+
+        Raises:
+            ApiError: the gate is not supported by the circuit parser.
+
+        """
+        del stream  # unused
```

Comment:
The signature of the method has to be the same as an existing gate to call it via the construction:
            gate_function = getattr(self, gate_name, getattr(self, "_gate_not_supported"))
            gate_function(stream, instruction)
in the parse methods. To prevent a warning from static code inspection I del't the unused parameters.

Reply:
I saw this comment in my email for some reason, and wanted to but in: I think `del`eting an unused parameter is a bit harsh. It is python convention to replace an unused parameter with a single underscore, `_`. This will shut up static code inspection and not destruct something of which you do not know if it is ok to destroy it.

Reply:
ok, good alternative I didn't think of.